src/Microsoft.Identity.Web/DownstreamApiSupport/DownstreamWebApi.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.Graph;
+using Microsoft.Identity.Web;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Implementation for the downstream API.
+    /// </summary>
+    public class DownstreamWebApi : IDownstreamWebApi
+    {
+        private readonly ITokenAcquisition _tokenAcquisition;
+        private readonly HttpClient _httpClient;
+        private readonly IOptionsMonitor<CalledApiOptions> _namedOptions;
+        private readonly JsonSerializerOptions _jsonOptions = new JsonSerializerOptions
+        {
+            PropertyNameCaseInsensitive = true,
+        };
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="tokenAcquisition">Token acquisition service.</param>
+        /// <param name="namedOptions">Named options provider.</param>
+        /// <param name="httpClient">Http client.</param>
+        public DownstreamWebApi(
+            ITokenAcquisition tokenAcquisition,
+            IOptionsMonitor<CalledApiOptions> namedOptions,
+            HttpClient httpClient)
+        {
+            _tokenAcquisition = tokenAcquisition;
+            _namedOptions = namedOptions;
+            _httpClient = httpClient;
+        }
+
+        /// <inheritdoc/>
+        public async Task<HttpResponseMessage> CallWebApiForUserAsync(
+            string optionsInstanceName,
```

Comment:
question: should we stick w/consistency of the naming? `MicrosoftIdentityCallWebApiForUserAysnc()`....i don't like this much, but i think things should feel like they are part of MS identity web, or not...again, will need to think about it.

Reply:
Here this is a verb (method) on a class that we own, not an extension method on a class that we don't own (like Service Collection). I don't think it's abiguous. I would stick to `CallWebApiForUserAsync` in this particular case